### DISCUSSION TITLE
Invalid type for argument in function call. Invalid implicit conversion from uint256 to bytes32 requested

### DISCUSSION THREAD

**Discussion by SuryaSundarVadali** [Cyfrin/foundry-full-course-cu#2160] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/393364cd-d62e-40e2-9d18-4d6769526d97)

Invalid type for argument in function call. Invalid implicit conversion from uint256 to bytes32 requested.solidity(9553)

**Comment by EngrPips** [comment]
Hello @SuryaSundarVadali, Please show your `Raffle` contract. Please don't show a picture, but instead copy the code and send it here. Make sure to format it nicely.

**Comment by SuryaSundarVadali** [comment]
`//SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

import {VRFConsumerBaseV2Plus} from "@chainlink/contracts/src/v0.8/vrf/dev/VRFConsumerBaseV2Plus.sol";
import {VRFV2PlusClient} from "@chainlink/contracts/src/v0.8/vrf/dev/libraries/VRFV2PlusClient.sol";
/**
 * @title Raffle
 * @author Surya
 * @notice creating a raffle contract
 * @dev This implements chainlink VRFv2.5
 */

contract Raffle is VRFConsumerBaseV2Plus {
    /*Errors*/
    error Raffle_NotEnoughETHSent();
    error Raffle_TransferFailed();
    error RaffleNotOpen();
    error Raffle_upKeepNotNeeded(uint256 balance,uint256 playersLength, uint256 raffleState);

    /*Type Declarations */
    enum RaffleState{
        OPEN,
        CALCULATING_WINNER
    } 
    /*State Variables*/ 
    uint16 private constant REQUEST_CONFRIMATIONS = 3;
    uint32 private constant NUM_WORDS = 1;
    bytes32 private immutable i_gasLane;
    uint256 private immutable i_subscriptionId;
    uint32 private immutable i_callbackGasLimit;

  
    uint256 private immutable i_entrance_fee;
    // @dev interval in seconds
    uint256 private immutable i_interval;
    address payable[] private s_players;
    uint256 private s_lastTimeStamp;
    address private i_vrfCoordinator;

    address private s_recentWinner;
    RaffleState private s_raffleState;

    event EnteredRaffle(address indexed player);
    event PickWinner(address indexed winner);

    constructor(
        uint256 entrance_fee,
        uint256 interval,
        address vrfCoordinator,
        bytes32 gasLane,
        uint256 subscriptionId,
        uint32 callbackGasLimit
    ) VRFConsumerBaseV2Plus(vrfCoordinator) {
        i_entrance_fee = entrance_fee;
        i_interval = interval;
        s_lastTimeStamp = block.timestamp;
        i_gasLane = gasLane;
        i_subscriptionId = subscriptionId;
        s_raffleState = RaffleState.OPEN;
        i_callbackGasLimit = callbackGasLimit;
    }

    function enterRaffle() external payable {
        if (msg.value <= i_entrance_fee) {
            revert Raffle_NotEnoughETHSent();
        }
        if(s_raffleState == RaffleState.CALCULATING_WINNER){
            revert RaffleNotOpen();
        }
        s_players.push(payable(msg.sender));
        emit EnteredRaffle(msg.sender);
    }

    function checkUpkeep(bytes memory /*check data*/) public view returns (bool upKeepNeeded, bytes memory /*perform memory*/){
        bool isOpen = s_raffleState == RaffleState.OPEN;
        bool timeHasPassed = (block.timestamp - s_lastTimeStamp) >= i_interval;
        bool hasBalance = address(this).balance > 0;
        bool hasPlayers = s_players.length > 0;
        upKeepNeeded = timeHasPassed && isOpen && hasBalance && hasPlayers;
        return (upKeepNeeded,"0x0");    
        }
    function performUpkeep(bytes calldata /* performData */) external{
        // check to see if enough time has passed
        (bool upKeepNeeded,) = checkUpkeep("");
        if(!upKeepNeeded){
            revert Raffle_upKeepNotNeeded(address(this).balance,s_players.length,uint256(s_raffleState));
        }
        s_raffleState = RaffleState.CALCULATING_WINNER;

        VRFV2PlusClient.RandomWordsRequest memory request = VRFV2PlusClient.RandomWordsRequest({
            keyHash: i_gasLane,
            subId: i_subscriptionId,
            requestConfirmations: REQUEST_CONFRIMATIONS,
            callbackGasLimit: i_callbackGasLimit,
            numWords: NUM_WORDS,
            extraArgs: VRFV2PlusClient._argsToBytes(VRFV2PlusClient.ExtraArgsV1({nativePayment: false}))
        });
        s_vrfCoordinator.requestRandomWords(request);
    }
    /*
     * @notice get the entrance fee
     * @return entrance fee
     */

    function fulfillRandomWords(uint256, uint256[] calldata randomWords) internal virtual override {
        uint256 indexOfWinner = randomWords[0] % s_players.length;
        address payable recentWinner = s_players[indexOfWinner]; 
        s_recentWinner = recentWinner;
        s_raffleState = RaffleState.OPEN;
        s_players = new address payable[](0);
        s_lastTimeStamp = block.timestamp;
        emit PickWinner(s_recentWinner);
        (bool success,) = recentWinner.call{value: address(this).balance}("");
        if(!success){
            revert Raffle_TransferFailed();
        }
        

    }

    function getEntranceFee() external view returns (uint256) {
        return i_entrance_fee;
    }
}
`

**Reply by EngrPips** [reply, marked as answer]
Hello @SuryaSundarVadali, Your constructor is below. 

```js
constructor(
    uint256 entrance_fee,
    uint256 interval,
    address vrfCoordinator,
    bytes32 gasLane,
    uint256 subscriptionId,
    uint32 callbackGasLimit
)
```
You see that your `Raffle` contract constructor. 

But in your `DeployRaffle` Clearly you swapped the `gasLane.`  and the `subscriptionId` position. So pass the argument in the correct order inside your `DeployRaffle` script, and you should get rid of that error.
```js
Raffle raffle = new Raffle(entrance_fee, interval, vrfCoordinator, subscriptionId, gasLane, callbackGasLimit);
```

**Reply by SuryaSundarVadali** [reply]
Thanks it worked

**Reply by EngrPips** [reply]
You are welcome. Keep crushing it.
